src/adapters/EasyRtcAdapter.js

```diff
@@ -173,58 +183,113 @@ class EasyRtcAdapter extends NoOpAdapter {
+        pendingMediaRequests.video.promise = videoPromise;
 
-        this.pendingMediaRequests.get(clientId).audio.promise = audioPromise;
-        this.pendingMediaRequests.get(clientId).video.promise = videoPromise;
+        this.pendingMediaRequests.set(clientId, pendingMediaRequests);
+      }
 
-        audioPromise.catch(e => NAF.log.warn(`${clientId} getMediaStream Audio Error`, e));
-        videoPromise.catch(e => NAF.log.warn(`${clientId} getMediaStream Video Error`, e));
+      const pendingMediaRequests = this.pendingMediaRequests.get(clientId);
+
+      // Create initial pendingMediaRequests with streamName
+      if (!pendingMediaRequests[streamName]) {
+        const streamPromise = new Promise((resolve, reject) => {
+          pendingMediaRequests[streamName] = { resolve, reject };
+        }).catch(e => NAF.log.warn(`${clientId} getMediaStream "${streamName}"" Error`, e))
+        pendingMediaRequests[streamName].promise = streamPromise;
       }
-      return this.pendingMediaRequests.get(clientId)[type].promise;
+
+      return this.pendingMediaRequests.get(clientId)[streamName].promise;
     }
   }
 
-  setMediaStream(clientId, stream) {
-    // Safari doesn't like it when you use single a mixed media stream where one of the tracks is inactive, so we
-    // split the tracks into two streams.
-    const audioStream = new MediaStream();
-    try {
-      stream.getAudioTracks().forEach(track => audioStream.addTrack(track));
-    } catch(e) {
-      NAF.log.warn(`${clientId} setMediaStream Audio Error`, e);
-    }
+  setMediaStream(clientId, stream, streamName) {
 
-    const videoStream = new MediaStream();
-    try {
-      stream.getVideoTracks().forEach(track => videoStream.addTrack(track));
-    } catch (e) {
-      NAF.log.warn(`${clientId} setMediaStream Video Error`, e);
-    }
+    const clientMediaStreams = this.mediaStreams[clientId] = this.mediaStreams[clientId] || new Map();
+    clientMediaStreams.set(streamName, stream);
 
-    this.mediaStreams[clientId] = { audio: audioStream, video: videoStream };
-
-    // Resolve the promise for the user's media stream if it exists.
     if (this.pendingMediaRequests.has(clientId)) {
-      this.pendingMediaRequests.get(clientId).audio.resolve(audioStream);
-      this.pendingMediaRequests.get(clientId).video.resolve(videoStream);
+      const pendingMediaRequests = this.pendingMediaRequests.get(clientId);
+
+      // Add mediaStreams audio streamName alias if does not exist yet
+      const audioTracks = stream.getAudioTracks();
+      if (!clientMediaStreams.has('audio') && audioTracks.length > 0) {
+        // Safari doesn't like it when you use single a mixed media stream where one of the tracks is inactive, so we
+        // split the tracks into two streams.
+        const audioStream = new MediaStream();
+        try {
+          audioTracks.forEach(track => audioStream.addTrack(track));
+          clientMediaStreams.set('audio', audioStream)
+        } catch(e) {
```

Comment:
I think this is wrong here. You should set the audioStream outside the `if (this.pendingMediaRequests.has(clientId))` like we had before. If I'm not mistaken, `setMediaStream` can be called before any `getMediaStream` (when the entity is not yet created, `networked-video-source` component didn't call `getMediaStream` yet), so in this case the `if (this.pendingMediaRequests.has(clientId))` will be false and will never set the audio and video.
This is probably the reason for the regression I have on my iPad.